test/lit/passes/remove-unused-module-elements-refs.wast

```diff
@@ -18,10 +18,16 @@
   ;; CHECK-NEXT:  (drop
   ;; CHECK-NEXT:   (ref.func $target-B)
   ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (call_ref $A
-  ;; CHECK-NEXT:   (ref.null $A)
+  ;; CHECK-NEXT:  (block
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (ref.null nofunc)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (unreachable)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT:  (block
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
```

Comment:
I think this PR makes the pass no longer test what it should,

https://github.com/WebAssembly/binaryen/blob/ddfebfa9b9014b80072ca77232181bd8e22c28ee/test/lit/passes/remove-unused-module-elements-refs.wast#L61-L62

We need an actual `call_ref` of the relevant type, but with this PR no such `call_ref` exists in IR even though there is one in the text.

If null doesn't work anymore, we could perhaps do a `ref.cast` of a null to that type? (Or we could use a `ref.func` to get that type, but that would take some more careful reading.)

Reply:
I'm going through and replacing a bunch of `ref.null`s with `local.get`s of new parameters to avoid introducing new `;; replaces something unreachable we can't emit` blocks.

Reply:
Pushed these changes now. Will look at the specific comments tomorrow.